flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/type/RowType.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.type;
+
+import org.apache.flink.types.Row;
+
+import java.util.Arrays;
+
+/**
+ * Row type for row.
+ *
+ * <p>It's internal data structure is BaseRow, and it's external data structure is {@link Row}.
+ */
+public class RowType implements InternalType {
+
+	/**
+	 * Use DataType instead of InternalType to convert to Row (if a Pojo in Row).
+	 */
+	private final InternalType[] types;
+
+	private final String[] fieldNames;
+
+	public RowType(InternalType... types) {
+		this(types, getFieldNames(types.length));
+	}
+
+	public RowType(InternalType[] types, String[] fieldNames) {
+		this.types = types;
+		this.fieldNames = fieldNames;
+	}
+
+	private static String[] getFieldNames(int length) {
```

Comment:
maybe change the name to `generateDefaultFieldNames`?

Reply:
Good suggestion